File: src/main/java/kr/co/dealmungchi/hotdealbatch/reactive/stream/RedisStreamMessage.java
package kr.co.dealmungchi.hotdealbatch.reactive.stream;

import lombok.Builder;
import lombok.Getter;

/**
 * Represents a message from a Redis stream.
 * Contains the message ID, stream key, provider, and encoded data.
 */
@Getter
@Builder
public class RedisStreamMessage {
    /**
     * The unique ID of the message in the Redis stream
     */
    private final String messageId;
    
    /**
     * The key of the Redis stream this message was read from
     */
    private final String streamKey;
    
    /**
     * The provider identifier (key in the Redis stream entry)
     */
    private final String provider;
    
    /**
     * The encoded data (value in the Redis stream entry)
     * This is typically Base64-encoded JSON
     */
    private final String data;
}

File: src/main/kotlin/kr/co/dealmungchi/hotdealbatch/reactive/stream/RedisStreamMessage.kt
package kr.co.dealmungchi.hotdealbatch.reactive.stream

/**
 * Represents a message from a Redis stream.
 * Contains the message ID, stream key, provider, and encoded data.
 */
data class RedisStreamMessage(
    /**
     * The unique ID of the message in the Redis stream
     */
    val messageId: String,
    
    /**
     * The key of the Redis stream this message was read from
     */
    val streamKey: String,
    
    /**
     * The provider identifier (key in the Redis stream entry)
     */
    val provider: String,
    
    /**
     * The encoded data (value in the Redis stream entry)
     * This is typically Base64-encoded JSON
     */
    val data: String
)